feat: Docs for react native touch events (#1785)
getsentry/sentry-docs: src/collections/_documentation/platforms/react-native/touchevents.md

---

> title: 'Tracking Touch Events'

Reviewer: what's about being a Breadcrumbs page instead of Touch events? I believe we'll be adding more of those and they would make sense to be aggregated IMO

Author: Hmm..yes it would make sense when we add more docs on breadcrumbs, but having this alone in there right now would be weird I'd think.

---

{% version_added 1.5.0 %}

> To track touch events with the React Native SDK you will need at least version `1.5.0` and will have to wrap your app with a `TouchEventBoundary`. This can be done in two ways:

Reviewer 2: What happens if multiple `TouchEventBoundaries` are used? Should we put a warning against that?

Author: You can actually have multiple if you want to wrap specific parts of the app, but they should not be nested. Good catch, will add that.

## Wrapping with `TouchEventBoundary`
At the root of your app, usually `App.js`, wrap the app component with `Sentry.TouchEventBoundary`:

```jsx
import * as Sentry from "@sentry/react-native";

const App = () => {
    return (
        <Sentry.TouchEventBoundary>
            <RestOfTheApp />
        </Sentry.TouchEventBoundary>
    );
}

export default AppRegistry.registerComponent("Your Amazing App", () => App);;
```

> ## Using the `withTouchEventBoundary` Wrapper

Author: Ah yes, definitely, I'll change it to use the HOC term.

At the root of your app, usually `App.js`, wrap the app component with `Sentry.withTouchEventBoundary`:

```jsx
import * as Sentry from "@sentry/react-native";

const App = () => {
    return (
        <RestOfTheApp />
    );
}

export default AppRegistry.registerComponent("Your Amazing App", () => Sentry.withTouchEventBoundary(App));;
```

# How Touches are Tracked

> Each touch event that occurs will be automatically logged as a breadcrumb, and will be displayed on the dashboard when an event occurs along with the component tree that the touch event occurred in.

Author: It actually uses `name` to unless there is a `displayName` present. Should make this more clear then.

## Naming Specific Components
You can let Sentry know which components to specifically track by setting the `displayName` property on them. If Sentry detects a component with a `displayName` within a touch's component tree, it will be logged nicely on the dashboard as having occurred in that component.
```jsx
class YourCoolComponent extends React.Component {
    static displayName = "CoolComponent";
    render() {
        return (
            <View>
                {/* ... */}
            </View>
        )
    }
}
```
or
```jsx
const YourCoolComponent = (props) => {
    return (
        <View>
            {/* ... */}
        </View>
    )
}

YourCoolComponent.displayName = "CoolComponent";
```

# Options

> Options can be passed to configure the boundary either as props to the `Sentry.TouchEventBoundary` component or as the second argument to the `Sentry.withTouchEventBoundary` wrapper.

Reviewer 3: ```suggestion
You can pass specific options to configure the boundary either as props to the `Sentry.TouchEventBoundary` component or as the second argument to the `Sentry.withTouchEventBoundary` wrapper.
```

```jsx
import * as Sentry from "@sentry/react-native";

const App = () => {
    return (
        <Sentry.TouchEventBoundary>
            <RestOfTheApp />
        </Sentry.TouchEventBoundary>
    );
}

export default AppRegistry.registerComponent("Your Amazing App", () => App);;
```

## Using the `withTouchEventBoundary` Higher-Order Component

At the root of your app, usually `App.js`, wrap the app component with `Sentry.withTouchEventBoundary`:

```jsx
import * as Sentry from "@sentry/react-native";

const App = () => {
    return (
        <RestOfTheApp />
    );
}

export default AppRegistry.registerComponent("Your Amazing App", () => Sentry.withTouchEventBoundary(App));;
```

{% capture __alert_content -%}
While you can track specific sections of your app by wrapping each section with the boundary, do not nest them or the touches could be tracked multiple times along with possible undefined behavior.
{%- endcapture -%}
{%- include components/alert.html
  title="Multiple Boundaries"
  content=__alert_content
  level="warning"
  deep_link="multiple-boundaries"
%}

## How Touches are Tracked Automatically

> Each touch event that occurs will be automatically logged as a breadcrumb, and will be displayed on the dashboard when an event occurs along with the component tree that the touch event occurred in. This component tree is logged using the `name` property of each component, which by default is the name given to it when it was defined or the name given to it by React Native such as "View" or "ScrollView".

Reviewer 2: ```suggestion
Each touch event that occurs will be automatically logged as a breadcrumb, and will be displayed on the dashboard when an event occurs along with the component tree that the touch event occurred in. This component tree is logged using the `name` property of a component. By default, React Native will set this property automatically on components.
```

> This component tree is logged using the `name` property of each component, which by default is the name given to it when it was defined or the name given to it by React Native such as "View" or "ScrollView".

Run on sentence, and we probably can just avoid going super detailed

As an aside did you use https://github.com/facebook/react-native/blob/3276563806966c16256181d9e71099f63f26e67f/Libraries/Renderer/implementations/ReactFabric-prod.js#L1020-L1060 when working on this?

Author: Ah no I didn't, I'll see if we can get access to that method. Thanks!
